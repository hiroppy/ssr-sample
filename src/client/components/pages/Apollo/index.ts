import { Apollo } from './Apollo';

export default Apollo;
