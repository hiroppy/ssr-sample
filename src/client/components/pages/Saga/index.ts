import { Saga } from './Saga';

export default Saga;
