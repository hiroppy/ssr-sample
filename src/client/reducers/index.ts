import { combineReducers } from 'redux';
import {
  reducer as pagesReducer,
  initialState as pagesInitialState,
  State as PagesState
} from './pages';
import {
  reducer as sagaPageReducer,
  initialState as sagaPageInitialState,
  State as SagaPageState
} from './sagaPage';

export type State = {
  pages: PagesState;
  sagaPage: SagaPageState;
};

export const initialState = {
  pages: pagesInitialState,
  sagaPage: sagaPageInitialState
};

export const rootReducer = combineReducers({
  pages: pagesReducer,
  sagaPage: sagaPageReducer
});
