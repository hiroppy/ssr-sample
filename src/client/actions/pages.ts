import { State } from '../reducers/pages';

export type Actions =
  | ReturnType<typeof setEnv>
  | ReturnType<typeof setBaseUrl>
  | ReturnType<typeof resetPageStatus>
  | ReturnType<typeof loadAppProcess>
  | ReturnType<typeof loadAppProcessSuccess>
  | ReturnType<typeof loadAppProcessFailure>
  | ReturnType<typeof loadTopPage>
  | ReturnType<typeof loadTopPageSuccess>
  | ReturnType<typeof loadTopPageFailure>
  | ReturnType<typeof loadSagaPage>
  | ReturnType<typeof loadSagaPageSuccess>
  | ReturnType<typeof loadSagaPageFailure>
  | ReturnType<typeof loadApolloPage>
  | ReturnType<typeof loadApolloPageSuccess>
  | ReturnType<typeof loadApolloPageFailure>;

export const SET_ENV = 'SET_ENV';
export const setEnv = (env: State['env']) =>
  ({
    type: SET_ENV,
    payload: {
      env
    }
  } as const);

export const SET_BASE_URL = 'SET_BASE_URL';
export const setBaseUrl = (baseUrl: State['baseUrl']) =>
  ({
    type: SET_BASE_URL,
    payload: {
      baseUrl
    }
  } as const);

export const RESET_PAGES_STATUS = 'RESET_PAGES_STATUS';
export const resetPageStatus = () =>
  ({
    type: RESET_PAGES_STATUS
  } as const);

export const LOAD_APP_PROCESS = 'LOAD_APP_PROCESS';
export const loadAppProcess = () =>
  ({
    type: LOAD_APP_PROCESS
  } as const);

export const LOAD_APP_PROCESS_SUCCESS = 'LOAD_APP_PROCESS_SUCCESS';
export const loadAppProcessSuccess = () =>
  ({
    type: LOAD_APP_PROCESS_SUCCESS
  } as const);

export const LOAD_APP_PROCESS_FAILURE = 'LOAD_APP_PROCESS_FAILURE';
export const loadAppProcessFailure = (err: Error) =>
  ({
    type: LOAD_APP_PROCESS_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_TOP_PAGE = 'LOAD_TOP_PAGE';
export const loadTopPage = () =>
  ({
    type: LOAD_TOP_PAGE
  } as const);

export const LOAD_TOP_PAGE_SUCCESS = 'LOAD_TOP_PAGE_SUCCESS';
export const loadTopPageSuccess = () =>
  ({
    type: LOAD_TOP_PAGE_SUCCESS
  } as const);

export const LOAD_TOP_PAGE_FAILURE = 'LOAD_TOP_PAGE_FAILURE';
export const loadTopPageFailure = (err: Error) =>
  ({
    type: LOAD_TOP_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_SAGA_PAGE = 'LOAD_SAGA_PAGE';
export const loadSagaPage = (maxLength: string | null) =>
  ({
    type: LOAD_SAGA_PAGE,
    payload: {
      maxLength
    }
  } as const);

export const LOAD_SAGA_PAGE_SUCCESS = 'LOAD_SAGA_PAGE_SUCCESS';
export const loadSagaPageSuccess = () =>
  ({
    type: LOAD_SAGA_PAGE_SUCCESS
  } as const);

export const LOAD_SAGA_PAGE_FAILURE = 'LOAD_SAGA_PAGE_FAILURE';
export const loadSagaPageFailure = (err: Error) =>
  ({
    type: LOAD_SAGA_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_APOLLO_PAGE = 'LOAD_APOLLO_PAGE';
export const loadApolloPage = () =>
  ({
    type: LOAD_APOLLO_PAGE
  } as const);

export const LOAD_APOLLO_PAGE_SUCCESS = 'LOAD_APOLLO_PAGE_SUCCESS';
export const loadApolloPageSuccess = () =>
  ({
    type: LOAD_APOLLO_PAGE_SUCCESS
  } as const);

export const LOAD_APOLLO_PAGE_FAILURE = 'LOAD_APOLLO_PAGE_FAILURE';
export const loadApolloPageFailure = (err: Error) =>
  ({
    type: LOAD_APOLLO_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);
