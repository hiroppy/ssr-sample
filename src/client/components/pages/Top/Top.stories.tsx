import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Top from '.';
import { initialState } from '../../../reducers';

export default {
  component: Top,
  title: 'Top'
};

const store = configureStore()(initialState);

export const Base = () => (
  <Provider store={store}>
    <Top />
  </Provider>
);
