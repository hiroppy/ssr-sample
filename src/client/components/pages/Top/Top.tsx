import React from 'react';
import { useDispatch } from 'react-redux';
import Md from 'react-markdown';
import styled from 'styled-components';
import raw from 'raw.macro';
import { loadTopPage } from '../../../actions/pages';
import { Head } from '../../Head';

const md = process.env.NODE_ENV !== 'test' ? raw('../../../../../README.md') : 'test-code';

const Container = styled.div`
  margin: auto;
  width: 80%;

  & pre {
    background: #333;
    color: #f5f5f5;
    padding: 4px 12px;
    overflow: auto;
  }
`;

export const Top: React.FC = () => {
  const dispatch = useDispatch();

  dispatch(loadTopPage());

  return (
    <>
      <Head title="top-page" />
      <section>
        <Container>
          <Md source={md} />
        </Container>
      </section>
    </>
  );
};
