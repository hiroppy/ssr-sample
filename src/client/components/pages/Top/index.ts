import { Top } from './Top';

export default Top;
