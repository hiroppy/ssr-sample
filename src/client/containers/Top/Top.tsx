import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { Top as TopComponent } from '../../components/pages/Top';
import { loadTopPage } from '../../actions/pages';
import { State } from '../../reducers';

const mapStateToProps = (state: State) => ({
  error: state.pages.error
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  load: () => {
    dispatch(loadTopPage());
  }
});

export const Top = connect(
  mapStateToProps,
  mapDispatchToProps
)(TopComponent);
