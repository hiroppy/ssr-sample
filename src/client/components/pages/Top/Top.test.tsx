import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Top from '.';
import { initialState } from '../../../reducers';
import * as actions from '../../../actions/pages';

test('should call loadTopPage via useDispatch', () => {
  const loadTopPage = jest.spyOn(actions, 'loadTopPage');
  const mockStore = configureStore()(initialState);

  render(
    <Provider store={mockStore}>
      <Top />
    </Provider>
  );

  expect(loadTopPage).toBeCalled();
});
