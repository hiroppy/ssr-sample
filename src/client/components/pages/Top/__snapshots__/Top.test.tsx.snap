// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and sub-components 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": false,
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "config": Object {
          "addTypename": false,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "clientAwareness": Object {},
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.5.1",
      "watchQuery": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <withProps(branch(branch(TopComponent)))
      error={null}
      load={[Function]}
    />
  </MemoryRouter>
</ApolloProvider>
`;
